modules/tasks/tasks/src/main/java/org/motechproject/tasks/service/TaskFilterExecutor.java

```diff
@@ -50,6 +53,7 @@ public TaskFilterExecutor() {
      */
     public boolean checkFilters(List<Filter> filters, LogicalOperator logicalOperator, TaskContext taskContext)
             throws TaskHandlerException {
+        LOGGER.debug("Checking if the task with name: {} matches the filters", taskContext.getTask().getName());
```

Comment:
Moreover we are swallowing an exception down below, that should also get logged on debug